### DISCUSSION TITLE
path parse parent

### DISCUSSION THREAD

**Discussion by adzubla** [nushell/nushell#8358] [Q&A]
Why parent includes "C:" if its is alreary in prefix?

```
〉'C:\Users\viking\spam.txt' | path parse
╭───────────┬─────────────────╮
│ prefix    │ C:              │
│ parent    │ C:\Users\viking │
│ stem      │ spam            │
│ extension │ txt             │
╰───────────┴─────────────────╯
```
Is it a bug?

**Comment by fdncred** [comment]
I'm not sure it's a bug, because it is part of the `parent`, but since we already have `prefix`, it kind of seems redundant maybe?

**Reply by adzubla** [reply]
Its strange because I would expect to compose a full path joining the components.

```
╭───────────┬─────────────────╮
│ prefix    │ C               │
│ parent    │ \Users\viking   │
│ stem      │ spam            │
│ extension │ txt             │
╰───────────┴─────────────────╯
```
With the above structure, I could do:

    $"($prefix):($parent)\($stem).($extension)"

Note that the prefix is only "C", the same as the extension is only "txt" and not ".txt"

**Reply by fdncred** [reply]
I think I'd want to have the `:` with the prefix. I've always considered the drive letter and colon the drive.

**Reply by ChrisDenton** [reply]
That is indeed how it's represented internally in the kernel.

<details><summary>the object manager</summary>

![screenshot from WinObjEx showing the `C:` drive as a symlink in the `\GLOBAL??` directory](https://chrisdenton.github.io/omnipath/images/Global.png)

</details>

**Reply by adzubla** [reply]
Makes sense, still the drive letter should not be part of the "parent" element. Example:

```
╭───────────┬─────────────────╮
│ prefix    │ C:              │
│ parent    │ \Users\viking   │
│ stem      │ spam            │
│ extension │ txt             │
╰───────────┴─────────────────╯

```

**Reply by adzubla** [reply]
It would be consisten with PowerShell:

```
PS> "C:\Users\viking\spam.txt" | Split-Path -Qualifier
C:
PS> "C:\Users\viking\spam.txt" | Split-Path -NoQualifier
\Users\viking\spam.txt
```

**Reply by fdncred** [reply]
We'd probably accept a PR for this. But being consistent with powershell is a non-goal, of course.
